packages/account-management/src/frontend/Registration/Registration.tsx

```diff
@@ -70,14 +70,8 @@ export const Registration: React.FC = () => {
   }, [email]);
 
   useEffect(() => {
-    // check if password passes password policy
-    if (pass && !passwordPolicy.test(pass || '')) {
-      setPassQualifies(false);
-      setCommonPassword(false);
-    } else {
-      setPassQualifies(true);
-      setCommonPassword(false);
-    }
+    setPassQualifies(!pass || passwordPolicy.test(pass || ''));
```

Comment:
On second thought, we can just do:

```suggestion
    setPassQualifies(passwordPolicy.test(pass));
```

`.test` can handle `undefined` and empty strings nae bother:

```
> const regex = /[A-Z]+/
undefined
> regex.test()
false
> regex.test('')
false
> regex.test('abc')
false
> regex.test('ABC')
true
```

Reply:
The issue here is about making sure we don't flag the password as invalid before the user has entered anything. If we make that change, the invalid password error message will be displayed from the moment the page loads

Reply:
That makes sense. In that case, this solution is a good one :+1: